reverseforwardrecord/get_dns_record.py

```diff
@@ -0,0 +1,255 @@
+    args, _ = ap.parse_known_args(inp_args)
+    return args
+
+
+def check_ip(reverse_order: bool = False,
+             output_filepath: str = "output.txt"):
+    """
+    checks for the ips and then appends them to lists to be checked on from later on in the script
+    :param reverse_order: reverse order tracks where to use the position in the function to be able
+    to correctly append the right string
+    :param output_filepath: filepath used to grab the ips from
+    :return: returns ips_found, dns_found
+    """
+    using_file = output_filepath
+    if not reverse_order:
+        position_dns = 0
+        position_ip = 2
+    else:
+        position_dns = 2
+        position_ip = 0
+    ips_found = []
+    dns_found = []
+    with open(using_file, 'r') as outfile:
+        for line in outfile.read().splitlines():
+            ips_found.append(line.split('\t')[position_ip])
+            dns_found.append(line.split('\t')[position_dns])
+    return ips_found, dns_found
+
+
+def is_reachable_ip(ip: str) -> bool:
+    """
+    used to ping an ip and see whether it is reachable or not
+    :param ip: converts ip to string
+    :return: returns boolean result
+    """
+    try:
+        result = subprocess.run(["ping", "-c", "1", ip], capture_output=True)
+        return result.returncode == 0
+    except Exception as e:
+        print(f"Error pinging IP {ip}: {e}")
+        return False
+
+
+def is_reachable_dns(dns_name: str) -> bool:
+    """
+    used to resolve a dns and see whether it is reachable or not
+    :param dns_name: converts dns to string
+    :return: returns boolean result
+    """
+    try:
+        socket.gethostbyname(dns_name)
+        return True
+    except socket.error as e:
+        print(f"Error resolving DNS {dns_name}: {e}")
+        return False
+
+
+def check_reachability(ips_found: List[str], dns_found: List[str]) \
+        -> Tuple[List[str], List[str], List[str], List[str]]:
+    """
+    checks whether the ips and dns names are reachable, using two previous functions and then
+    appending the results of those to the lists which are after displayed to the command line
+    :params ips_found: ips in a list to be tested
+    :params dns_found: dns names in a list to be tested
+    :params tuple_lists: lists that cant be changed
+    :return: all the ips found and dns names found that into 2 sections being reachable meaning
+    working ips or dns or unreachable which are out of use ips or dns that cannot be reached
+    """
+    reachable_ips = []
+    unreachable_ips = []
+    reachable_dns = []
+    unreachable_dns = []
+
+    for ip in ips_found:
+        if is_reachable_ip(ip):
+            reachable_ips.append(ip)
+        else:
+            unreachable_ips.append(ip)
+
+    for dns_name in dns_found:
+        if is_reachable_dns(dns_name):
+            reachable_dns.append(dns_name)
+        else:
+            unreachable_dns.append(dns_name)
```

Comment:
This is more efficient 

```suggestion
    reachable_ips = list(filter(is_reachable_ip, ips_found))
    unreachable_ips = [ip for ip in ips_found if ip not in reachable_ips]

    reachable_dns = list(filter(is_reachable_dns, dns_found))
    unreachable_dns = [dns for dns in dns_found if dns not in reachable_dns]
```

Reply:
done